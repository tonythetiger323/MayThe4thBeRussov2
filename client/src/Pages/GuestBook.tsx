import React from 'react'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { Image, Transformation } from 'cloudinary-react'
import Paper from '@material-ui/core/Paper'
import MessageDialog from '../Components/MessageDialog'
import { withStyles, Theme } from '@material-ui/core/styles'
import GuestbookMessages from '../Components/GuestBookMessages'
import Responsive from 'react-responsive'

const Mobile = (props: any) => <Responsive {...props} maxWidth={667} />
const Default = (props: any) => <Responsive {...props} minWidth={668} />

const styles = (theme: Theme) => ({
  root: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(2)
  }
})

function GuestBook(props: any) {
  const { classes } = props

  return (
    <div>
      <Grid
        container
        direction="column"
        alignItems="center"
        alignContent="center"
        justify="center"
      >
        <Grid item>
          <Typography variant="h4" align="center">
            Guest Book
          </Typography>
        </Grid>
        <br />
        <Grid item>
          <Default>
            <Image
              cloudName="tonythetiger323"
              publicId="MayThe4thBeRusso/Other Photos/Together/FullSizeRender.jpg"
            >
              <Transformation width="550" crop="scale" />
            </Image>
          </Default>
          <Mobile>
            <Image
              cloudName="tonythetiger323"
              publicId="MayThe4thBeRusso/Other Photos/Together/FullSizeRender.jpg"
            >
              <Transformation width="375" crop="scale" />
            </Image>
          </Mobile>
        </Grid>
        <br />
        <Paper style={{ opacity: 0.6 }} className={classes.root}>
          <Grid
            container
            direction="column"
            alignItems="center"
            alignContent="center"
            justify="center"
          >
            <Grid item>
              <Typography
                style={{ fontFamily: 'Oswald' }}
                variant="h6"
                align="center"
              >
                Help us remember this very special time in our lives by leaving
                us a note! Thank you so much for celebrating with us!
              </Typography>
            </Grid>
            <br />
            <Grid item>
              <MessageDialog />
            </Grid>
          </Grid>
        </Paper>
        <br />
        <Grid item>
          <Typography
            variant="h4"
            style={{ fontFamily: 'Oswald' }}
            align="center"
          >
            Guest Book Messages
          </Typography>
        </Grid>
        <br />
        <Grid item>
          <GuestbookMessages />
        </Grid>
      </Grid>
    </div>
  )
}

export default withStyles(styles)(GuestBook)
