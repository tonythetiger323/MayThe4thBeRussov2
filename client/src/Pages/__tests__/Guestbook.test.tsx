import * as React from 'react'
import { shallow } from 'enzyme'
import GuestBook from '../GuestBook'

describe('Guest Book Page', () => {
  let wrapper: any
  beforeEach(() => (wrapper = shallow(<GuestBook />)))

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
