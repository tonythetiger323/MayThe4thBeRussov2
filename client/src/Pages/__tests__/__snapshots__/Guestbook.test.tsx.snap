// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guestbook Page should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Guestbook />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Typography)>
         Leave a Message for Megan & Randy!
      </WithStyles(Typography)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": " Leave a Message for Megan & Randy!",
      },
      "ref": null,
      "rendered": " Leave a Message for Megan & Randy!",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Typography)>
           Leave a Message for Megan & Randy!
        </WithStyles(Typography)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": " Leave a Message for Megan & Randy!",
        },
        "ref": null,
        "rendered": " Leave a Message for Megan & Randy!",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
